tests/test_NoiseReplacer.py

```diff
@@ -75,15 +76,76 @@ def setUp(self):
+        task = self.makeSingleFrameMeasurementTask("test_NoiseReplacer")
+        task.config.noiseReplacer.noiseSource = 'measure'
+        exposure, catalog = self.dataset.realize(1.0, task.schema, randomSeed=0)
+        task.run(catalog, exposure)
+        sumVariance = exposure.getMaskedImage().getVariance().getArray().sum()
+        for record in catalog:
+            self.assertFloatsAlmostEqual(record.get("test_NoiseReplacer_inside"),
+                                         record.get("truth_instFlux"), rtol=1E-3)
+
+            # N.B. Next line checks that a random value is correct to a
+            # statistical 1-sigma prediction; some RNG seeds may cause it to
+            # fail (indeed, 67% should)
+            self.assertLess(record.get("test_NoiseReplacer_outside"), np.sqrt(sumVariance))
 
+    def testVarianceNoiseReplacer(self):
+        """Test noise replacement in SFM with ''variance'' mode.
+        """
+        # We choose a random seed which causes the test to pass.
+        task = self.makeSingleFrameMeasurementTask("test_NoiseReplacer")
+        task.config.noiseReplacer.noiseSource = 'variance'
+        exposure, catalog = self.dataset.realize(1.0, task.schema, randomSeed=0)
+        task.run(catalog, exposure)
+        sumVariance = exposure.getMaskedImage().getVariance().getArray().sum()
+        for record in catalog:
+            self.assertFloatsAlmostEqual(record.get("test_NoiseReplacer_inside"),
+                                         record.get("truth_instFlux"), rtol=1E-3)
+
+            # N.B. Next line checks that a random value is correct to a
+            # statistical 1-sigma prediction; some RNG seeds may cause it to
+            # fail (indeed, 67% should)
+            self.assertLess(record.get("test_NoiseReplacer_outside"), np.sqrt(sumVariance))
+
+    def testMetaNoiseReplacer(self):
+        """Test noise replacement in SFM with ''meta'' mode.
+        """
+        # We choose a random seed which causes the test to pass.
+        task = self.makeSingleFrameMeasurementTask("test_NoiseReplacer")
+        task.config.noiseReplacer.noiseSource = 'meta'
+        var = 2.0
+        exposure, catalog = self.dataset.realize(np.sqrt(var),
+                                                 task.schema,
+                                                 randomSeed=0)
+        md = PropertyList()
+        md['BGMEAN'] = var
+        exposure.setMetadata(md)
+        task.run(catalog, exposure)
+        sumVariance = exposure.getMaskedImage().getVariance().getArray().sum()
+        for record in catalog:
+            self.assertFloatsAlmostEqual(record.get("test_NoiseReplacer_inside"),
+                                         record.get("truth_instFlux"), rtol=1E-3)
+
+            # N.B. Next line checks that a random value is correct to a
+            # statistical 1-sigma prediction; some RNG seeds may cause it to
+            # fail (indeed, 67% should)
```

Comment:
OK, so this comment claims that 67% should fail statistically, but there's actually been no fine tuning there - all the random seeds are zero, and all apparently pass. What's up with that? (Yes, it could be luck, but still, worth wondering here).